### PR TITLE
add mixed demo

### PR DESCRIPTION
**Pull request by zhfnjust**

### CODE REVIEW

File: tests/js/arraydemo.scrypttest.js [hunk 1]
```diff
@@ -41,27 +101,46 @@ describe('Test sCrypt contract ArrayDemo In Javascript', () => {
     });
 
 
-    // it('should return true', () => {
-    //     result = arraydemo.testArrayInt([1, 1321, 243213, 32132]).verify()
-    //     expect(result.success, result.error).to.be.true
-    // });
+    it('should return true', () => {
```
Comment (xhliu) [inline tests/js/arraydemo.scrypttest.js:104]: Some public functions are not tested, such as `unlock()`. Let's cover all.

File: tests/js/arraydemo.scrypttest.js [hunk 2]
```diff
@@ -1,13 +1,16 @@
 const { expect, assert } = require('chai');
-const { buildContractClass, Ripemd160, Sig } = require('scryptlib');
+const { buildContractClass, Ripemd160, Sig, Bytes, buildTypeClasses } = require('scryptlib');
 const { compileContract } = require('../../helper');
 
+const compileResult = compileContract('arraydemo.scrypt');
+const ArrayDemo = buildContractClass(compileResult);
+const { ST1, ST2, ST3, MDArrayST1, AliasST2 } = buildTypeClasses(compileResult);
```
Comment (xhliu) [inline tests/js/arraydemo.scrypttest.js:7]: Let's add `buildTypeClasses` into `scryptlib's` README, so ppl can know how to use it.

**Review by xhliu — APPROVED** [review]
🥇 very nice.


Make sure u address all comments before merging.

File: contracts/arraydemo.scrypt [hunk 3]
```diff
@@ -1,9 +1,38 @@
+
```
Comment (xhliu) [inline contracts/arraydemo.scrypt:1]: Rename to `arrayStructDemo.scrypt`

File: tests/js/arraydemo.scrypttest.js (hunk 1 above)
Comment (zhfnjust) [reply tests/js/arraydemo.scrypttest.js]: add

**Pull request merged**